File: apps/ChoreUI/src/app/entities/overdue-chore.ts
import { Chore } from './chore';
import { Contributor } from './contributor';

export class OverdueChore {
  Chore: Chore 
  Contributor: Contributor;
  EventDate: Date;
  ExpectedEventDate: Date;
  OverdueDays: number;

  public static CreateFromDTO(data: any): OverdueChore {
    if (data == null)
    {
      return null;
    }

    let chore = new OverdueChore();
    chore.Chore = Chore.CreateFromDTO(data);
    chore.Contributor = Contributor.CreateFromDTO(data);
    chore.EventDate = data.EventDate;
    chore.ExpectedEventDate = data.ExpectedEventDate;
    chore.OverdueDays = data.OverdueDays;

    return chore;
  }
}

File: apps/ChoreAPI/Overdue/OverdueChore.cs
using ChoreHelper.WebApi.Chores;
using ChoreHelper.WebApi.Contributors;
using System;

namespace ChoreHelper.WebApi.Overdue
{
    public class OverdueChore 
    {
        public Chore Chore { get; set; }

        public Contributor Contributer { get; set; }

        public DateTime? EventDate { get; set; }

        public DateTime? ExpectedEventDate { get; set; }

        public int? OverdueDays { get; set; }
    }
}
